tests/azure_iot_ut.c

```diff
@@ -0,0 +1,172 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <setjmp.h>
+
+#include <cmocka.h>
+
+#include "azure_iot.h"
+/*-----------------------------------------------------------*/
+static const uint8_t ucTestKey[] = "Hello";
+static const uint8_t ucTestValue[] = "World";
+
+static uint8_t ucBuffer[1024];
+
+/*-----------------------------------------------------------*/
+
+uint32_t ulGetAllTests();
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesInit_Failure( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+    
+    ( void ) ppvState;
+
+    /* Fail init when null Message control block */
+    assert_int_not_equal( AzureIoT_MessagePropertiesInit( NULL, ucBuffer, 0, sizeof( ucBuffer ) ),
+                          eAzureIoTSuccess );
+
+    /* Fail init when null Buffer */
+    assert_int_not_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                          NULL, 0, 0 ),
+                          eAzureIoTSuccess );                                     
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesInit_Success( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );  
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesAppend_Failure( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucTestKey ) ),
+                      eAzureIoTSuccess );
+
+    /* Failed for NULL key passed */
+    assert_int_not_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
```

Comment:
Nit: Could we check for the actual expected error?  This would help pin behavior of what we return in various cases and if this changes, the UT would trip over it and we could have a discussion RE backcompat concerns if any.